modules/puppet_connect/plans/test_input_data.pp

```diff
@@ -0,0 +1,31 @@
+# @summary
+#   Tests that the provided Puppet Connect input data is complete, meaning that all consuming inventory targets are connectable.
+#
+# This plan should only be used as part of the copy-pastable "test input data"
+# workflow specified in the Puppet Connect docs.
+#
+# @param targets
+#   The inventory targets. This should never be set by the user. It is only here for compatibility with Bolt's rerun workflow.
+#
+# @return ResultSet the result of invoking the 'is connectable?' query on all
+# the targets. Note that this query currently consists of running the 'whoami'
+# command.
+#
+plan puppet_connect::test_input_data(TargetSpec $targets = 'all') {
+  $targs = get_targets($targets)
+  $targs.each |$target| {
+    if $target.transport != 'ssh' and $target.transport != 'winrm' {
+      fail_plan("Inventory contains target ${target} with unsupported transport, must be ssh or winrm")
+    }
+    if $target.host == undef {
+      fail_plan("Inventory contains invalid target ${target}, the target's 'host' key is unknown")
+    }
+    if $target.transport == 'ssh' {
+      # Disable SSH autoloading to prevent false positive results
+      # (input data is wrong but target is still connectable due
+      # to autoloaded config)
+      set_config($target, ['ssh', 'load-config'], false)
```

Comment:
We can set this unconditionally. If the target isn't using ssh, it will just ignore the ssh portion of the config.

Reply:
Actually on second thought, I'd rather leave this as-is because it is more in-line with how you'd explain it to someone ("we set load-config to false for SSH targets" => "if target is ssh, set load-config to false"). Knowing that it will still do the right thing for WinRM targets is a nice implementation detail, but an implementation detail that doesn't result in significant cleanup.